gensim/similarities/docsim.py

```diff
@@ -937,7 +937,7 @@ def __init__(self, corpus, similarity_matrix, num_best=None, chunksize=256, norm
         """
         self.similarity_matrix = similarity_matrix
 
-        self.corpus = corpus
+        self.corpus = list(corpus)
```

Comment:
This needs a change in the documentation too IMO – a big warning that this is happening, the implications.

Reply:
Isn't this arguably the expected behavior? The documentation of SoftCosineSimilarity does not make it clear what happens to the constructur, but isn't it assumed that the corpus is read, indexed, and stored inside the SimilarityABC class?

Reply:
If this is not the expected behavior, then there is no grounds for this PR in the first place.

Reply:
On second thoughts, you're right.
It is a reasonable expectation that the indexing process will a) traverse the corpus, and b) grow in in RAM with corpus size.